lib/src/features/user/ui/perf_stats_screen.dart

```diff
@@ -0,0 +1,536 @@
+
+    final dateFormatter = DateFormat.yMMMd(Intl.getCurrentLocale());
+
+    return game == null
+        ? const Text('?', style: defaultDateStyle)
+        : Text(dateFormatter.format(game!.finishedAt), style: defaultDateStyle);
+  }
+}
+
+class _RatingWidget extends StatelessWidget {
+  final int? rating;
+  final UserPerfGame? game;
+  final Color color;
+
+  const _RatingWidget(this.rating, this.game, this.color);
+
+  @override
+  Widget build(BuildContext context) {
+    return (rating == null)
+        ? const Text('?', style: TextStyle(fontSize: _defaultValueFontSize))
+        : Column(
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              Text(rating.toString(),
+                  style:
+                      TextStyle(fontSize: _defaultValueFontSize, color: color)),
+              _UserGameWidget(game)
+            ],
+          );
+  }
+}
+
+class _PercentageValueWidget extends StatelessWidget {
+  final int value;
+  final int denominator;
+  final Color? color;
+  final bool isShaded;
+
+  const _PercentageValueWidget(this.value, this.denominator,
+      {this.color, this.isShaded = false});
+
+  String _getPercentageString(num numerator, num denominator) {
+    return '${((numerator / denominator) * 100).round()}%';
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final trueColor = color ?? DefaultTextStyle.of(context).style.color;
+
+    return Column(mainAxisAlignment: MainAxisAlignment.center, children: [
+      Text(value.toString(),
+          style: TextStyle(fontSize: _defaultValueFontSize, color: trueColor)),
+      Text(_getPercentageString(value, denominator),
+          style: TextStyle(
+              fontSize: _defaultValueFontSize,
+              color: isShaded
+                  ? trueColor!.withOpacity(_customOpacity / 2)
+                  : trueColor!.withOpacity(_customOpacity)))
+    ]);
+  }
+}
+
+class _StreakWidget extends StatelessWidget {
+  final UserStreak? maxStreak;
+  final UserStreak? curStreak;
+  final Color? color;
+
+  const _StreakWidget(this.maxStreak, this.curStreak, {this.color});
+
+  @override
+  Widget build(BuildContext context) {
+    final trueColor = color ?? DefaultTextStyle.of(context).style.color;
+    final valueStyle =
+        TextStyle(color: trueColor, fontSize: _defaultValueFontSize);
+
+    final streakTitleStyle = TextStyle(
+        fontSize: _defaultStatFontSize,
+        color: textShade(context, _customOpacity));
+
+    final List<Widget> streakWidgets = List.empty(growable: true);
+    final longestStreakStr = context.l10n.longestStreak('').replaceAll(':', '');
+    final currentStreakStr = context.l10n.currentStreak('').replaceAll(':', '');
+    // This variable may seem useless, why not check (streak == maxStreak)?
+    // Because this avoids the rare case where maxStreak == curStreak and the same title is shown twice.
+    bool inFirstLoop = true;
+
```

Comment:
There's nothing wrong with the way you've written it, but for the sake of example here's a more functional approach, using the `Iterable.mapIndexed` method of flutter collection lib: https://api.flutter.dev/flutter/package-collection_collection/IterableExtension/mapIndexed.html

```dart
final List<Widget> = [maxStreak, curStreak].mapIndexed((index, streak) {
  final streakTitle = Text(index == 0 ? longestStreakStr : currentStreakStr);
  ...
}).toList(growable: false);
```
I general I prefer the functional style because:
1. you don't need to mutate a list
2. no need for the mutable `inFirstLoop` variable

Sometimes I also write in the imperative style, but when I can I'd always choose the more functional style.